src/analysisd/decoders/syscollector.c

```diff
@@ -1756,3 +1911,105 @@ int decode_process(Eventinfo *lf, cJSON * logJSON,int *socket) {
     free(msg);
     return retval;
 }
+
+const char** get_field_list(const char *type) {
+    char const **ret_val = NULL;
+    if (NULL != type) {
+        if (strcmp(type, "hotfixes") == 0) {
+            ret_val = hotfixes_fields;
+        } else if(strcmp(type, "packages") == 0) {
+            ret_val = packages_fields;
+        } else if(strcmp(type, "processes") == 0) {
+            ret_val = processes_fields;
+        } else if(strcmp(type, "ports") == 0) {
+            ret_val = ports_fields;
+        } else if(strcmp(type, "network_iface") == 0) {
+            ret_val = network_iface_fields;
+        } else if(strcmp(type, "network_protocol") == 0) {
+            ret_val = network_protocol_fields;
+        } else if(strcmp(type, "network_address") == 0) {
+            ret_val = network_address_fields;
+        } else if(strcmp(type, "hardware") == 0) {
+            ret_val = hardware_fields;
+        } else if(strcmp(type, "os") == 0) {
+            ret_val = os_fields;
+        } else {
+            merror("Incorrrect/unknown type value %s.", type);
+            ret_val = false;
+        }
+    } else {
+        merror("null type value.");
+        ret_val = false;
+    }
+    return ret_val;
+}
+
+bool fill_data_dbsync(cJSON *data, const char *field_list[], char *msg) {
+    int ret_val = false;
+
+    while (NULL != *field_list) {
+        cJSON *key = cJSON_GetObjectItem(data, *field_list);
+        if (NULL != key) {
+            if (NULL != cJSON_IsNumber(key)) {
+                char value[OS_SIZE_128] = { 0 };
+                snprintf(value, OS_SIZE_128 - 1, "%d", key->valueint);
+                strcat(msg, value);
+            } else if (NULL != cJSON_IsString(key)) {
+                if(strlen(key->valuestring) == 0) {
+                    strcat(msg, "NULL");
+                } else {
+                    strcat(msg, key->valuestring);
+                }
+            } else {
+                strcat(msg, "NULL");
+            }
+        } else {
+            strcat(msg, "NULL");
+        }
+        strcat(msg, "|");
```

Comment:
I'd suggest adding a comment here explaining, briefly, why it's necessary cut this message by "|" character.